eng/pipelines/docindex.yml

```diff
@@ -2,3 +2,44 @@ trigger: none
 
 jobs:
   - template: /eng/common/pipelines/templates/jobs/docindex.yml
+
```

Comment:
Is this eventually going to go into tools/common?

Reply:
These changes are written to be minimally differentiated and easy to put into a template. We should always have a platform specific `docindex.yml` file so we can opt repos into or out of docs.ms publishing processes.